sql/src/main/java/io/crate/execution/package-info.java

```diff
@@ -0,0 +1,164 @@
+ *             \             /  (modulo based distribution based on the group by key)
+ *              \___________/   (aggregation to "partial")
+ *             /            \
+ *            /              \
+ *     MergePhase           MergePhase         // reducer
+ *          |     _________/    (aggregation to "final")
+ *          |    /
+ *     MergePhase                              // localMerge
+ *           (concat results; apply final limit)
+ * </pre>
+ *
+ * <p>
+ *     This involves the following components:
+ * </p>
+ *
+ * <pre>
+ *     - On the handler the {@link io.crate.execution.engine.ExecutionPhasesTask} is used to:
+ *       - Create a local {@link io.crate.execution.jobs.JobExecutionContext}, this contains:
+ *          - A {@link io.crate.execution.engine.collect.JobCollectContext}
+ *          - Two {@link io.crate.execution.jobs.PageDownstreamContext}, one for each MergePhase
+ *       - Fire a {@link io.crate.execution.jobs.transport.JobRequest} to n2
+ *
+ *     - {@link io.crate.execution.jobs.transport.TransportJobAction} will receive this request and based on it create
+ *        - A {@link io.crate.execution.engine.collect.JobCollectContext}
+ *        - One {@link io.crate.execution.jobs.PageDownstreamContext}
+ *
+ *     - {@link io.crate.execution.jobs.JobExecutionContext#start()} triggers the start method on the sub-contexts which will:
+ *
+ *       - Initiate the collect operation
+ *       (This means a RowConsumer, in this case a {@link io.crate.execution.engine.distribution.DistributingConsumer}, will receive a {@link io.crate.data.BatchIterator};
+ *       consume it's data and forward it's result.
+ *
+ *       - The data is received by {@link io.crate.execution.engine.distribution.TransportDistributedResultAction}
+ *       which looks up a {@link io.crate.execution.jobs.PageDownstreamContext} to process it.
+ *
+ *       This fills up a {@link io.crate.data.BatchIterator} with the data that was received and also invokes a {@link io.crate.data.RowConsumer}.
+ *       Again a {@link io.crate.execution.engine.distribution.DistributingConsumer}
+ *
+ *       - The data is again received by a {@link io.crate.execution.jobs.PageDownstreamContext} (MergePhase on the handler).
+ *         The data is again being made available to a {@link io.crate.data.RowConsumer} via a {@link io.crate.data.BatchIterator}.
+ *         The RowConsumer here is a {@link io.crate.action.sql.RowConsumerToResultReceiver}, which forwards the result to a client.
+ *
+ *
+ *       Visually, this may look as follows:
+ *
+ *       ## n1
+ *
+ *              (src: LuceneBatchIterator)
+ *       CollectingBatchIterator (does the grouping)
+ *          |
+ *       DistributingConsumer
+ *          |
+ *          .
+ *          . Network I/O
+ *          |
+ *       TransportDistributedResultAction (also receives results from n2)
+ *          |
+ *       PageDownstreamContext
+ *          |
+ *          |   (src: BatchPagingIterator)
```

Comment:
This skips when this iterator is created. In the PageDownstreamContext, no?

Reply:
Not sure if I get what you mean. Can you clarify?

Reply:
Is the BatchPagingIterator created in the PageDownStreamContext?

Reply:
Currently yes, but I wanted to avoid going into that much detail, if you zoom in you'll see that there are several more components involved. This presentation here is a simplified view.

Should I add a comment to clarify that (that it's a simplified representation) ?

Reply:
Yes, I reckon that there is more to it. I just wonder how helpful it is to have the source information available because it left me wondering how the BatchpagingIterator is created. So I would suggest removing the source information or adding more details about how the BatchPagingIterator is constructed.